proto/storage/auth_provider.proto

```diff
@@ -35,4 +35,14 @@ message AuthProvider {
     }
     // EXPERIMENTAL.
     repeated RequiredAttribute required_attributes = 11;
+
+    // EXPERIMENTAL.
+    // visibility allows to specify whether the auth provider should show up when listing the auth providers
+    // during login.
+    enum Visibility {
+        UNSET                    = 0;
+        VISIBLE                  = 1;
+        NOT_VISIBLE              = 2;
```

Comment:
super nit: I would call it `HIDDEN` to highlight that visible is the default (i.e., hiding it is what requires an active choice)